### PR TITLE
Code refactor / TGUI-ification of the VR Explosive Simulator

### PR DESCRIPTION
**Pull request by MeggalBozale**
## About the PR 
Improves the quality of the code for the VR bomb simulator, and gives it a snazzy new look!
To not go overboard, I'll have outputting the log through the simulator as a separate PR (Did you know it output a log? I didn't!) But I do intend to add that after TGUI-ification.

Before/after (Panel closed)
dreamseeker_MKrxZsLaO4
image

Before/after (Panel open)
dreamseeker_EF3NFzfToU
YwDM7SsSv1

## Why's this needed? 
Improving code quality and using TGUI windows is generally better, as TGUI windows improve the UI/UX and improving code quality helps make future changes when desired.

https://github.com/orgs/goonstation/projects/7/views/1?pane=issue&itemId=53462121

## Changelog
```
(u)MeggalBozale
(*)TGUI interface and minor functional changes have been added to the VR Explosive Simulator.
```

Can atomize the PR to split TGUI changes to another PR if this is too much, just LMK if so

### CODE REVIEW

**Comment by MeggalBozale** [comment]
This is in a functional state but still needs testing and scrutiny. I'll be up tomorrow to polish and tweak as need be.

**Comment by ZeWaka** [comment]
Screenshot would be sick

**Comment by MeggalBozale** [comment]
Tested the following scenarios:
Putting in tanks by clicking the sim while inhand
Putting in tanks by clicking the buttons while inhand
Ejecting with hands empty, current hand empty, both hands full
Ejecting and re-inserting repeatedly in various configurations

Starting sim, and then ejecting tanks
Starting sim, and attempting to add tanks while both are full

Starting sim with an explosive mix
Starting sim with an inert mix (simulation stays in progress until it ends via time limit, can start sim again after cooldown)

Putting butts in tank one/two and starting, ending simulation (because why not allow buttbomb testing)
(fixed a few runtimes for that)

Reviewing generated logs for maxcap (Works good)
Reviewing generated logs for buttbomb
Reviewing generated logs for inert
buttbomb+inert WORK, but run into a max file size limit. I'll handle that in a future PR for improved logging.
(Issue is with cat /mnt/mixer/bomblog, contents are all there -- so getting it for a TGUI readout should be easy.)

Putting bombs in after each other also worked fine.

**Comment by MeggalBozale** [comment]
Fixed issues with tanks rupturing in the machine causing runtimes, Im not sure what else there is to test

File: code/obj/item/tank.dm [hunk 1]
```diff
@@ -316,6 +316,18 @@ Contains:
...
+	icon_state = "empty"
+	var/custom_gas_type = "oxygen" // replace with exact varname of choice
+	var/custom_temperature = 175 KELVIN
+
+	New()
+		..()
+		src.air_contents.temperature = src.custom_temperature
+		src.air_contents.vars[src.custom_gas_type] = (10 * ONE_ATMOSPHERE) * src.air_contents.volume / (R_IDEAL_GAS_EQUATION * src.air_contents.temperature)
+
+
```
Comment (TobleroneSwordfish) [inline code/obj/item/tank.dm:330]: What dis for?

File: tgui/packages/tgui/interfaces/Bombsim/index.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+ * @author Romayne (https://github.com/MeggalBozale)
+ * @license ISC
+ */
+
+import { useBackend, useLocalState } from '../../backend';
+import { Button, LabeledList, Section, Stack } from '../../components';
+import { Window } from '../../layouts';
+import { formatPressure, truncate } from '../../format';
+
+const TankDisplay = (props, context) => {
```
Comment (TobleroneSwordfish) [inline tgui/packages/tgui/interfaces/Bombsim/index.tsx:13]: Why not just use the tank display that TTVs/canisters use, for parity

File: code/modules/networks/computer3/mainframe2/misc_terms.dm [hunk 3]
```diff
@@ -735,303 +738,284 @@ TYPEINFO(/obj/machinery/networked/storage)
...
+
+#define TANK_ONE "1"
+#define TANK_TWO "2"
+/// Checks if we have a tank in the specified slot
+#define HAS_TANK(tanknum) ((tanknum == TANK_ONE) && src.tank1) || ((tanknum == TANK_TWO) && src.tank2)
+/// Add the tank to the slot being interact with in the device
+#define ADD_TANK(tanknum, tank) tank.set_loc(src); if ((tanknum) == TANK_ONE) {src.tank1 = tank;}\
+								else {src.tank2 = tank;};
+	/// Interact with one of the tank slots on the machine. "1" for the first slot, "2" for the second. "null" to use an empty slot, if there is one.
+	proc/interact_tank_slot(mob/user, obj/item/I, slot=null)
```
Comment (TobleroneSwordfish) [inline code/modules/networks/computer3/mainframe2/misc_terms.dm:764]: Consider manually forcing a TGUI update here to make it a little more responsive.

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
Looks good, a couple of a minor things.

File: code/obj/item/tank.dm (hunk 1 above)
Comment (MeggalBozale) [reply code/obj/item/tank.dm:330]: woops forgot to remove that. custom tank was used for testing with atlas so I could just throw a maxcap mix in and not varedit it 20 times. dont know how i missed it, thank u

File: tgui/packages/tgui/interfaces/Bombsim/index.tsx (hunk 2 above)
Comment (MeggalBozale) [reply tgui/packages/tgui/interfaces/Bombsim/index.tsx:13]: ill attempt that and post how it looks :+1:

**Comment by MeggalBozale** [comment]
Changes made, new look is shown at top but i will paste it here also
YwDM7SsSv1

**Review by TobleroneSwordfish — APPROVED** [review]

**Comment by Sovexe** [comment]
instead of a + for the begin explosion button icon you could have something fun like `bomb`

**Comment by MeggalBozale** [comment]
image

**Pull request merged**